src/bp/migrations/v12_17_2-1612973566-add_botid_to_dialog_sessions_id_and_remove_botid_column.ts

```diff
@@ -0,0 +1,63 @@
+import * as sdk from 'botpress/sdk'
+import { Migration } from 'core/services/migration'
+
+const TABLE_NAME = 'dialog_sessions'
+const TEMP_TABLE_NAME = 'dialog_sessions_temp'
+
+const migration: Migration = {
+  info: {
+    description: 'Merge botId column into dialog_sessions id',
+    target: 'core',
+    type: 'database'
+  },
+  up: async ({
+    bp,
+    configProvider,
+    database,
+    inversify,
+    metadata
+  }: sdk.ModuleMigrationOpts): Promise<sdk.MigrationResult> => {
+    const db = database.knex as sdk.KnexExtended
+    const { client } = db.client.config
+
+    try {
+      if (client === 'sqlite3') {
+        await db.transaction(async trx => {
+          await trx.raw(`UPDATE ${TABLE_NAME} SET id = botId || '::' || id;`)
```

Comment:
I'm not sure what you're trying to do here ... you set the id to being the botId ? what's the logic behind that. Or perhaps I don't understand that statement.

I think what we're trying to do is (pseudo) :  `id = id + botId`

Reply:
I choose to _prefix_ the session id with the botId. So, in the migration, I first update the table ids with the botId like so `id = botId || '::' || id` which is the equivalent of  `id = botId + separator + id`

As for testing, I just made sure that the ids were prefixed with the botId, that the botId column was dropped and that the id column was the primary key (for both SQLite and PostgreSQL)

Reply:
Wow, totally sorry I'm used to  use the concat function and had no idea of the || operator .. that is totally my bad I thought you were making `botID or '::"` which makes no sens